### PR TITLE
add notes for n18418

### PR DESCRIPTION
**Pull request by glozow**
Recommended music pairing: https://www.youtube.com/watch?v=-33tz2b10Vc

### CODE REVIEW

File: _posts/2021-05-19-#18418.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+host: glozow
+status: upcoming
+commit: 241e141
+---
+
+## Notes
+
+* The way our wallet constructs transactions over time can leak information
+  about its contents.  The most obvious example is we can assume that all UTXOs
+to the same address are controlled by the same person. UTXOs sent to different
```
Comment (jnewbery) [inline _posts/2021-05-19-#18418.md:17]: "all UTXOs to the same address" -> "all UTXOs sent to the same address"/"all UTXOs with the same address"/"all UTXOs with the same scriptPubKey"

File: _posts/2021-05-19-#18418.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+address.  Another is grouping UTXOs into
+[OutputGroup](https://github.com/bitcoin-core-review-club/bitcoin/blob/4ac1adda9914d845aaea5804af4801ffec53c701/src/wallet/coinselection.h#L72)s
+by scriptPubKey and spending them together.
+
+* However, `OutputGroup`s can grow quite large - after all, anyone can send as
+  many transactions as they want to an address that we've published. It might
+not make sense to fund a 0.015BTC transaction by sweeping a group of 150 inputs
+worth 10BTC (not to mention the extra fees for all the unnecessary inputs).
+The `OUTPUT_GROUP_MAX_ENTRIES` constant limits the number of UTXOs per
+`OutputGroup`. Within `GroupOutputs()`, if a scriptPubKey exceeds this many, we
```
Comment (jnewbery) [inline _posts/2021-05-19-#18418.md:36]: "if a scriptPubKey exceeds this many" -> "if we have more than `OUTPUT_GROUP_MAX_ENTRIES` UTXOs with the same scriptPubKey"

File: _posts/2021-05-19-#18418.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+batch them into multiple `OutputGroup`s with up to `OUTPUT_GROUP_MAX_ENTRIES`
+UTXOs each.
+
+* [PR#18418](https://github.com/bitcoin/bitcoin/pull/18418) increases
+  `OUTPUT_GROUP_MAX_ENTRIES` from 10 to 100. The number 100 was suggested
+during a previous review club,
+[#17824](https://bitcoincore.reviews/17824.html#l-339) This behavior change
```
Comment (jnewbery) [inline _posts/2021-05-19-#18418.md:43]:
```suggestion
[#17824](https://bitcoincore.reviews/17824.html#l-339). This behavior change
```

Comment (jnewbery) [inline _posts/2021-05-19-#18418.md:43]: Perhaps:

```suggestion
during [a previous review club](https://bitcoincore.reviews/17824.html#l-339). This behavior change
```

File: _posts/2021-05-19-#18418.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+constitutes just one line (and some adjustments to the tests), but it is ripe
+with opportunities to explore how coin selection works.
+
+* For beginners, host recommends simply adding log statements and (don't forget
```
Comment (jnewbery) [inline _posts/2021-05-19-#18418.md:47]: "host recommends" is a little bit awkward. Perhaps:

Try adding some log statements, re-compiling and then re-running the tests (Hint: test/functional/combine_logs.py to see logs, and you can assert that your logs are printed by adding with node.assert_debug_log(expected_msg=[your_log_statement]) to the functional test).

File: _posts/2021-05-19-#18418.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  to re-compile) then re-running the tests (Hint:
+`test/functional/combine_logs.py` to see logs, and you can assert that your
+logs are printed by adding `with
+node.assert_debug_log(expected_msg=[your_log_statement])` to the functional
+test).  Some good tests to play around with are wallet\_avoidreuse.py and
+wallet\_groups.py.  [This doc](https://github.com/fjahr/debugging_bitcoin) is a
+great resource (also authored by fjahr)!  You can also tinker some of the
```
Comment (jnewbery) [inline _posts/2021-05-19-#18418.md:54]:
```suggestion
wallet\_groups.py.  The PR author, fjahr, has written an excellent [guide to debugging Bitcoin Core](https://github.com/fjahr/debugging_bitcoin) with some hints on adding logging and using debuggers. You can also tinker some of the
```

File: _posts/2021-05-19-#18418.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+	(A) `avoid_reuse = true, avoidpartialspends = true`
+
+	(B) `avoid_reuse = true, avoidpartialspends = false`
+
+	(C) `avoid_reuse = false, avoidpartialspends = true`
+
+	(D) `avoid_reuse = false, avoidpartialspends = false`
+
+
+3. If your wallet has 101 UTXOs of 0.01 BTC each and tries to send a payment of
```
Comment (jnewbery) [inline _posts/2021-05-19-#18418.md:82]: Does this need to state that the 101 UTXOs are to the same address?

**Review by jnewbery — COMMENTED** [review]
Great notes! A few suggestions inline.

File: _posts/2021-05-19-#18418.md (hunk 6 above)
Comment (glozow) [reply _posts/2021-05-19-#18418.md:82]: woops! good point

**Comment by glozow** [comment]
Sorry for the delay - took all your comments @jnewbery

**Pull request merged**